### ISSUE TITLE
Spurious error casting Comparable.compare from Comparable<dynamic> to Comparable<T>

### ISSUE THREAD

**Issue by jacob314** [dart-lang/dev_compiler#23]
```dart
/// A [Comparator] that uses a [FastFieldAccessor] to speed up comparisons based
/// on nested fields in an [Entity].
class FastComparator<S extends Entity, T> {

  final FastFieldAccessor<S, T> accessor;
  final Comparator<T> comparator;

  FastComparator(String field) : this.withComparator(field,  
  * /* DDC:severe: StaticTypeError, Type check failed: Comparable.compare ((Comparable<dynamic>, Comparable<dynamic>) → int) is not of type (T, T) → int */ 
    Comparable.compare);

  FastComparator.withComparator(String field, this.comparator) :
      accessor = new FastFieldAccessor<S, T>(field);

  int call(S a, S b) => comparator(accessor[a], accessor[b]);
}
```

**Comment by jmesserly** [comment]
it sounds like we landed in an okay place here
